charts/ratify/templates/assign.yaml

```diff
@@ -0,0 +1,89 @@
+{{- if semverCompare ">= 3.11.0" .Values.gatekeeper.version }}
+apiVersion: mutations.gatekeeper.sh/v1
+{{- else }}
+apiVersion: mutations.gatekeeper.sh/v1beta1
+{{- end }}
+kind: Assign
+metadata:
+  name: mutate-pod-image
+spec:
+  match:
+    scope: Namespaced
+    kinds:
+    - apiGroups: ["*"]
+      kinds: ["Pod"]
+    excludedNamespaces:
+      - "gatekeeper-system"
+      {{- if ne .Release.Namespace "gatekeeper-system"}}
+      - {{ .Release.Namespace }}
+      {{- end }}
+  applyTo:
+  - groups: [""]
+    kinds: ["Pod"]
+    versions: ["v1"]
+  location: "spec.containers[name:*].image"
+  parameters:
+    assign:
+      externalData:
+        provider: ratify-mutation-provider
+
+---
+{{- if semverCompare ">= 3.11.0" .Values.gatekeeper.version }}
+apiVersion: mutations.gatekeeper.sh/v1
+{{- else }}
+apiVersion: mutations.gatekeeper.sh/v1beta1
+{{- end }}
+kind: Assign
+metadata:
+  name: mutate-workload-image
+spec:
+  match:
+    scope: Namespaced
+    kinds:
+    - apiGroups: ["apps", "batch"]
+      kinds: ["Deployment", "ReplicaSet", "StatefulSet", "DaemonSet", "Job"]
```

Comment:
Can these be pooled? `apps` api group is only valid for `Deployment`, `ReplicaSet`, `StatefulSet` and `DaemonSet`. `Job` is `batch`. Also, do you need to support `CronJobs`?

Reply:
Is mutation done only per pod or high level resource? If it's only pods, then just `mutate-pod-image` is sufficient.

Reply:
mutation is currently done for the deployment resource and at the pod resource level. We don't know if user will use workload resource or not which is why I added both